exchanges/gateio/gateio_wrapper.go

```diff
@@ -2172,21 +2173,110 @@ func (g *Gateio) UpdateOrderExecutionLimits(ctx context.Context, a asset.Item) e
 				minBaseAmount = math.Pow10(-int(pairsData[x].AmountPrecision))
 			}
 
-			limits = append(limits, order.MinMaxLevel{
-				Asset:                   a,
-				Pair:                    pair,
+			l = append(l, limits.MinMaxLevel{
+				Key:                     key.NewExchangePairAssetKey(g.Name, a, pair),
 				QuoteStepIncrementSize:  math.Pow10(-int(pairsData[x].Precision)),
 				AmountStepIncrementSize: math.Pow10(-int(pairsData[x].AmountPrecision)),
 				MinimumBaseAmount:       minBaseAmount,
 				MinimumQuoteAmount:      pairsData[x].MinQuoteAmount.Float64(),
 			})
 		}
+	case asset.Futures:
+		btcContracts, err := g.GetAllFutureContracts(ctx, currency.BTC)
+		if err != nil {
+			return err
+		}
+		usdtContracts, err := g.GetAllFutureContracts(ctx, currency.USDT)
+		if err != nil {
+			return err
+		}
+		btcContracts = append(btcContracts, usdtContracts...)
+		l = make([]limits.MinMaxLevel, 0, len(btcContracts))
+		for x := range btcContracts {
+			if btcContracts[x].InDelisting {
+				continue
+			}
+			p := strings.ToUpper(btcContracts[x].Name)
+			if !g.IsValidPairString(p) {
+				continue
+			}
+			cp, err := currency.NewPairFromString(p)
+			if err != nil {
+				return err
+			}
+			l = append(l, limits.MinMaxLevel{
+				Key:                     key.NewExchangePairAssetKey(g.Name, a, cp),
+				MinimumBaseAmount:       float64(btcContracts[x].OrderSizeMin),
+				MaximumBaseAmount:       float64(btcContracts[x].OrderSizeMax),
+				MarketStepIncrementSize: btcContracts[x].OrderPriceRound.Float64(),
+				AmountStepIncrementSize: btcContracts[x].OrderPriceRound.Float64(),
```

Comment:
This looks like it should be `PriceStepIncrementSize` and below

Reply:
I previously had them set to 1 (as in 1 contract) but thought I'd try and link it to something better in anticipation of your comments 😭

Reply:
Yeah I set them as 1 in my own stuff and the step as 1 if its for contracts. Can't do this for every exchange though, because some (bybit I think?) uses the actual base amounts instead of contract and multi. If you do go down that route, maybe have a futures package const ContractStep = 1 maybe as an idea.